### PR TITLE
Array macros

### LINKED ISSUE #307: [CT-1159] Migrate array cross-db macros from dbt-utils
### Describe the feature
See https://github.com/dbt-labs/dbt-core/issues/5520

### PR DESCRIPTION
**Pull request by dbeatty10**
resolves #307
related to https://github.com/dbt-labs/dbt-core/issues/5520

### Description

All the related pull requests:
- https://github.com/dbt-labs/dbt-core/pull/5823
- https://github.com/dbt-labs/dbt-redshift/pull/182
- https://github.com/dbt-labs/dbt-snowflake/pull/257
- https://github.com/dbt-labs/dbt-bigquery/pull/308
- https://github.com/dbt-labs/dbt-spark/pull/454

### Checklist

- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-bigquery/blob/main/CONTRIBUTING.md#Adding-CHANGELOG-Entry)

### CODE REVIEW

**Comment by dbeatty10** [comment]
This is ready for review, and it depends on https://github.com/dbt-labs/dbt-core/pull/5823.

After https://github.com/dbt-labs/dbt-core/pull/5823 is approved and merged, I will revert the changes to `dev-requirements.txt` before merging this PR.

File: dev-requirements.txt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # install latest changes in dbt-core
 # TODO: how to automate switching from develop to version branches?
-git+https://github.com/dbt-labs/dbt-core.git#egg=dbt-core&subdirectory=core
-git+https://github.com/dbt-labs/dbt-core.git#egg=dbt-tests-adapter&subdirectory=tests/adapter
+git+https://github.com/dbt-labs/dbt-core.git@dbeatty/lift-shift-array-macros#egg=dbt-core&subdirectory=core
```
Comment (colin-rogers-dbt) [inline dev-requirements.txt:3]: remove this change

Comment (dbeatty10) [reply dev-requirements.txt:3]: See explanation [here](https://github.com/dbt-labs/dbt-bigquery/pull/308#issuecomment-1254396335):

> This is ready for review, and it depends on [dbt-labs/dbt-core#5823](https://github.com/dbt-labs/dbt-core/pull/5823).
> 
> After [dbt-labs/dbt-core#5823](https://github.com/dbt-labs/dbt-core/pull/5823) is approved and merged, I will revert the changes to `dev-requirements.txt` before merging this PR.

**Review by colin-rogers-dbt — APPROVED** [review]
LGTM

**Pull request merged**